lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    # === In ManageIQ/manageiq-ui-classic PR #1234:
+    #
+    # The following command:
+    #
+    #   @miq-bot run-tests manageiq#5678
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=manageiq#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=manageiq-ui-classic#1234
+    #
+    class RunTest < Base
+      # The user calling the command
+      attr_reader :issuer
+
+      # The org of the issue this command originated from
+      attr_reader :org
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.github.run_tests_repo.url
+      end
+
+      def self.test_repo_name
+        Settings.github.run_tests_repo.name
+      end
+
+      def self.test_repo_clone_dir
+        @test_repo_clone_dir ||= begin
+                                   url_parts = test_repo_url.split("/")[-2, 2]
+                                   repo_org  = url_parts.first
+                                   repo_dir  = test_repo_name
+                                   File.join(::Repo::BASE_PATH, repo_org, repo_dir)
+                                 end
+      end
+
+      # Name of the bot
+      def self.bot_name
+        Settings.github_credentials.username
+      end
+
+      # The new branch name for this particular run of the command (uniq)
+      def branch_name
+        @branch_name ||= "#{self.class.bot_name}-run-tests-#{issue.number}-#{SecureRandom.uuid}"
+      end
+
+      def run_tests
+        clone_repo_if_necessary
+        create_cross_repo_test_branch
+        update_travis_yaml_content
+        commit_travis_yaml_changes
+        push_commit_to_remote
+        create_cross_repo_test_pull_request
+      end
+
+      private
+
+      def _execute(issuer:, value:)
+        @issuer = issuer
+        parse_value(value)
+        return unless valid?
+
+        run_tests
+      end
+
+      def parse_value(value)
+        @value = value
+
+        @test_repos, @repos = value.split(/\s+including\s+/)
+                                   .map { |repo_list| repo_list.split(",") }
+
+        # Add the identifier for the PR for this comment to @repos here
```

Comment:
Something about this wording feels missing...perhaps:

```ruby
# Ensure the identifier for the PR itself is added to the list of repos
```

On that note, could this end up duplicating the repo if it's already in the list?

Reply:
Yeah, maybe we should use a `Set` here, huh.

I guess using a normalized `repo_name` does have a purpose @agrare :)